docs/under-the-hood/storage.md

```diff
@@ -1,138 +1,201 @@
-
-Check the reference guide to [read more about temporal
-types](/reference-guide/data-types.md) `Duration`, `Date`, `LocalTime` and
-`LocalDateTime`. 
-
-:::note
-
-Even though it's possible to store `List` and `Map` property values, it is
-impossible to modify them. It is, however, possible to replace them entirely.
-So, the following queries are valid:
-
-```cypher
-CREATE (:Node {property: [1, 2, 3]});
-CREATE (:Node {property: {key: "value"}});
-```
-
-But these are not:
-
-```cypher
-MATCH (n:Node) SET n.property[0] = 0;
-MATCH (n:Node) SET n.property.key = "other value";
-```
+All properties use at least **1B** for encoding the extra information about the
+type of the value stored, the size of the property key (is an integer stored in
+1B, 2B, 4B, or 8B), and the last **4bits** are used for a variety of reasons
+depending on the value type of the property. After the initial byte, properties
+store the value of its key.
 
-:::
+Minimal size without values: **2-9B**
+
+#### Size of the value
+
+|Value                                  |Size                                                    |
+|---------------------------------------|--------------------------------------------------------|
+|`NULL`                                 |1B                                                      |
+|`BOOL`                                 |1B                                                      |
+|`INT` uses the minimal amount of bytes |1B, 2B, 4B or 8B                                        |
+|`DOUBLE`                               |8B                                                      |
+|`STRING`                               |min 2B                                                  |
+|`LIST`                                 |                                                        |
+|`MAP`                                  |                                                        |
+|`TEMPORAL DATA`                        |8B                                                      |
+
+### Marvel dataset use case
+
+The Marvel dataset consists of `Hero`, `Comic` and `ComicSeries` labels, which are indexed. There are also three label-property indices - on the `name` property of `Hero` and `Comic` vertices, and on the `title` property of `ComicSeries` vertices. The `ComicSeries` vertices also have the `publishYear` property.
+
+ <img src={require('../data/under-the-hood/marvel-dataset-schema.png').default}/>
+
+There are 6487 `Hero` and 12661 `Comic` vertices with the property `name`. That's 19148 vertices in total. To calculate how much storage those vertices and properties occupy, we are going to use the following formula:
+
+$\texttt{NumberOfVertices} \times (\texttt{Vertex} + \texttt{properties} + \texttt{SkipListNode} + \texttt{next\_pointers} + \texttt{Delta}).$
+
+Let's assume the name on average has $2\text{B}+10\text{B} = 12\text{B}$ (each name is on average 10 characters long). When we include the average values, we get:
```

Comment:
@katarinasupe Not sure how you got info about each char size and string in the end but ok, maybe we should check this one with Buda. This things about string can vary a lot, because of encoding support etc..

Reply:
This was Buda's calculation. I only added the calculation for the property `publishYear` that was missing.